ansible/slurm-drain.yml

```diff
@@ -0,0 +1,38 @@
+---
+- import_playbook: cluster-infra-query.yml
+
+- hosts: openstack
+  gather_facts: false
+  tasks:
+    - name: Count the number of compute nodes per slurm partition
+      set_fact:
+        desired_state: "{{ ((cluster_group.output_value | selectattr('group', 'equalto', item.name ) | first).nodes | map(attribute='name') | list )[:item.num_nodes] + desired_state | default([]) }}"
+      with_items: "{{ openhpc_slurm_partitions }}"
+    - debug: var=desired_state
+
+- hosts:
+  - cluster_control
+  become: yes
+  tasks:
+  - name: Get nodes in DRAINED state
+    command: "sinfo --noheader --Node --format='%N' --states=DRAINED"
+    register: drained_nodes
+    changed_when: false
+  - name: Get nodes in ALLOC,IDLE states
+    command: "sinfo --noheader --Node --format='%N' --states=ALLOC,IDLE"
+    register: resumed_nodes
+    changed_when: false
+
+- hosts:
+  - cluster_batch
+  become: yes
+  roles:
+    - role: stackhpc.openhpc
+      openhpc_cluster_name: "{{ cluster_name }}"
+      openhpc_slurm_control_host: "{{ groups['cluster_control'] | first }}"
+      openhpc_enable:
+        control: false
```

Comment:
These new parameters are a little fishy - we are kind of implying a deployment without anything in it, which is a counterintuitive way of having the effect we want (which is not to deploy anything).  Thankfully, *ahem* this mode of operation's going to be clearly documented in the role's README...

Reply:
I'm not 100% clear about what you mean here.